src/VisualStudio/CSharp/Impl/Options/CSharpSettingsManagerOptionSerializer.cs

```diff
@@ -274,7 +296,67 @@ public override bool TryPersist(OptionKey optionKey, object value)
                 }
             }
 
+            // code style: use var options.
+            if (optionKey.Option == CSharpCodeStyleOptions.UseVarForIntrinsicTypes)
+            {
+                return PersistUseVarOption(Style_UseVarForIntrinsicTypes, value);
+            }
+            else if (optionKey.Option == CSharpCodeStyleOptions.UseVarWhenTypeIsApparent)
+            {
+                return PersistUseVarOption(Style_UseVarWhenTypeIsApparent, value);
+            }
+            else if (optionKey.Option == CSharpCodeStyleOptions.UseVarWherePossible)
+            {
+                return PersistUseVarOption(Style_UseVarWherePossible, value);
+            }
+
             return base.TryPersist(optionKey, value);
         }
+
+        private bool PersistUseVarOption(string option, object value)
+        {
+            var convertedValue = (SimpleCodeStyleOption)value;
+            var offset = convertedValue.IsChecked ? 0 : Enum.GetValues(typeof(DiagnosticSeverity)).Length;
+            var baseValue = (int)convertedValue.Notification.Value;
+
+            this.Manager.SetValueAsync(option, value: baseValue + offset, isMachineLocal: false);
+            return true;
+        }
+
+        private static bool FetchUseVarOption(int useVarOptionValue, out object value)
+        {
+            switch (useVarOptionValue)
+            {
+                case 0:
```

Comment:
Oh man... do these two switches need to be in sync?  If so, please let's not do this.  Let's just have a string representation, and we can have SImpleCodeStyleOption serialize/deserialize itself from that.